### DISCUSSION TITLE
Reading data from a referenced table via API.

### DISCUSSION THREAD

**Discussion by enterrift76** [supabase/supabase#21001] [Questions] [answered]
Good day everyone,

I have two tables, one is "Vehicles" and one is "Drivers". I connect the drivers table to the vehicle table by adding the "id" in the Drivers table as a primary key in the vehicles table. When I perform a GET to list information from a specific vehicle, the driver is returned as the "id" stored in the table (this makes perfect sense). What would be the best practice if I want to return the driver's name instead of id in the vehicle table?

**Comment by GaryAustin1** [comment, marked as answer]
You have a foreign key from vehicles to to driver id primary key.
Then you can just use `select('*,Drivers(name))')`
https://supabase.com/docs/guides/database/joins-and-nesting

**Reply by GaryAustin1** [reply]
You really should not switch to different questions from the title as follow on users will get confused or miss out on info in searches.

You likely would use cron extension to schedule a repeating task.  That task would use the http or pg_net extension to read your API and store the data.  Those are all covered lightly in the docs and are standard Postgres things so lots of info on the web.

**Reply by enterrift76** [reply]
Noted, deleted & marked the answer.
